### ISSUE TITLE
Server version cannot be compiled on FreeBSD, GOPATH issue

### ISSUE THREAD

**Issue by herrbischoff** [dnote/dnote#246]
* project/version (e.g. CLI v0.8.0): Server
* operating system: FreeBSD 12.0 RELEASE

## Description

Server version cannot be compiled on FreeBSD, complains about GOPATH.

## Steps to reproduce

The `Makefile` appears to be a GNU Make file, so I'm using `gmake`.

```
$ gmake
==> installing go dependencies
both /home/herrbischoff/dnote-server-v0.1.0-beta and /usr/home/herrbischoff/dnote-server-v0.1.0-beta are not within any known GOPATH
gmake: *** [Makefile:28: install-go] Error 1
```

This is correct, since:

```
$ pwd
/home/herrbischoff/dnote-server-v0.1.0-beta
```

and 

```
echo $GOPATH
/home/herrbischoff/go
```

**Comment by sungwoncho** [comment]
Thanks for the detailed instruction. It seems like your working directory is not in`$GOPATH`.

* Could you clone the project under `$GOPATH/src/github.com/dnote/dnote` and try again? You can do so by running `go get github.com/dnote/dnote`.
* Also, the build script is configured to build a binary for linux/amd64. You might need to add something like `build freebsd amd64` to the following

https://github.com/dnote/dnote/blob/master/pkg/server/scripts/build.sh#L60

Here are all possible `GOOS` and `GOARCH` values: https://stackoverflow.com/questions/20728767/all-possible-goos-value

I can also consider distributing the server binary for FreeBSD. One thing that is stopping me is that I could not figure out how to compile Dnote CLI for FreeBSD due to https://github.com/karalabe/xgo/issues/91.

**Comment by herrbischoff** [comment]
Thanks for the pointers, however:

```
$ go get github.com/dnote/dnote
package github.com/dnote/dnote: no Go files in /home/herrbischoff/go/src/github.com/dnote/dnote
```

Those clones the repository, albeit with the warning.

```
$ cd go/src/github.com/dnote/dnote/
$ gmake
==> installing go dependencies
==> installing js dependencies
cd: /src/github.com/dnote/dnote/web: No such file or directory
gmake: *** [Makefile:41: install-js] Error 2
```

There's an absolute path there, no doubt because the Makefile does not use the set `GOPATH`. However `GOPATH` is set:

```
$ echo $GOPATH
/home/herrbischoff/go
```

Replacing all occurrences of `${GOPATH}` with `/home/herrbischoff/go` works:

```
$ gmake
==> installing go dependencies
==> installing js dependencies
npm WARN karma-webpack@3.0.5 requires a peer of webpack@^2.0.0 || ^3.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN dnote@1.0.0 No repository field.
npm WARN dnote@1.0.0 license should be a valid SPDX license expression
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"freebsd","arch":"x64"})

audited 15276 packages in 12.595s
found 354 high severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
```

The amount of known vulnerabilities is rather alarming and should be mitigated by updating the dependencies to newer versions. I'd be happy to help with this as soon as I get this compiled and running.

Then again, this:

```
$ gmake version=v0.1.0 build-server
==> building web
/bin/sh: ./scripts/build-prod.sh: not found
gmake: *** [Makefile:77: build-web] Error 127
```

And indeed, there's no `build-prod.sh`. There are several other scripts in the Makefile targets that are not in the repo.

**Comment by sungwoncho** [comment]
@herrbischoff thanks for your patience. I was able to reproduce your problem in FreeBSD 12 and fixed it. Your comment outlines three problems:

1. `GOPATH` environment variables is not evaluated in the Makefile.
2. `./scripts/build-prod.sh` seems to be missing.
3. There are a number of security vulnerabilities in the JavaScript dependencies.

**Problem 1**

I was not able to reproduce the problem 1. At the moment, I am not too sure why this is happening because GNU Make references says:

> either ‘$(foo)’ or ‘${foo}’ is a valid reference to the variable foo
> https://www.gnu.org/software/make/manual/html_node/Reference.html#Reference
> Every environment variable that make sees when it starts up is transformed into a make variable with the same name and value.
> https://www.gnu.org/software/make/manual/html_node/Environment.html

**problem 2**

The error message seems to be saying `/bin/sh/` is missing, rather than the file itself. The file indeed exists [here](https://github.com/dnote/dnote/blob/master/web/scripts/build-prod.sh). I have merged #249 to fix it. You will need `bash` to run some of the scripts.

**problem 3**

I have upgraded the dependencies to apply security patch in #248. In the long run, I will try to automate this process.

---

All in all, you will be able to build the server project. I still welcome your thoughts on the first problem, though.

By the way, `gmake version=v0.1.0 build-server` will likely fail because the version number contains the letter 'v'. Please try `gmake version=0.1.0 build-server`

**Comment by herrbischoff** [comment]
Thanks, I copied the `version=v0.1.0` string from your instructions in the Makefile. So you should probably change them there.

With the shebang changes, the project builds. However. I still needed to add the environment variable explicitly: `env GOPATH=/Users/mbischoff/go make version=0.1.0 build-server` Must be some quirk related to Fish shell, which I use.

Thanks for your help, this solves this for me for the time being.
